pkg/mic/mic.go

```diff
@@ -275,6 +369,13 @@ func (c *Client) Sync(exit <-chan struct{}) {
 		begin := time.Now()
 		workDone := false
 
+		cacheTime := time.Now()
+
+		// There is a delay in data propogation to cache. It's possible that the creates performed in the previous sync cycle
+		// are not propogated before this sync cycle began. In order to avoid redoing the cycle, we sync cache again.
+		c.CRDClient.SyncCacheAll(exit, false)
```

Comment:
Were you able to confirm that this cache sync with the go client update has no leaks?